org-ql-completing-read.el

```diff
@@ -265,7 +265,23 @@ single predicate)."
       (mapc #'org-ql--ensure-buffer buffers-files)
       (let* ((completion-styles '(org-ql-completing-read))
              (completion-styles-alist (list (list 'org-ql-completing-read #'try #'all "Org QL Find")))
-             (selected (completing-read prompt #'collection nil t)))
+             (selected
+              (minibuffer-with-setup-hook
+                  (lambda ()
+                    (let ((map (make-sparse-keymap)))
+                      (define-key
+                       map (kbd "C-c C-e")
```

Comment:
@oantolin Two questions:

1. Should we use a predefined keymap?
2. Should we borrow the appropriate, equivalent Embark binding, in case the user's rebound it?

Thanks for your patience.  I think this is the last thing I will merge before tagging v0.8 of org-ql.

Reply:
1. The command bound to `C-c C-e` is not fixed, since it is a lexical closure capturing the `buffers-files` variable. If the idea behind question 1 about using a predefined keymap is to make the key configurable I'd suggest instead a user option containing the desired key binding. We have a few similar user options in Embark and Consult. Is that why you suggested the keymap?

2. Well, `embark-export` is not globally bound to anything usually, and in `embark-general-map` it is bound to `E`, which we certainly don't want here (since `E` needs to self-insert). I guess you could try the binding `C-. E`, but that's awkward for people who do bind `C-.` to `embark-act`, as suggested in the Embark documentation. I think on the balance the best binding is the `C-c C-e` I chose.

Reply:
@oantolin Well, I was hoping that we could find a way to display the binding in the Embark menu, like all the other Embark bindings relevant to the `org-ql-find` commands.  I'd prefer if users didn't have to remember the binding to use it, nor see it in the documentation in order to realize it exists.  (And I'll probably forget it, myself, if it's not in the menu.)  Is there an easy way to do that?

Reply:
I added a new commit, 0fd7045, that adds a keymap and a fake `org-ql-export-view` command. This command just prints an error message if run outside `org-ql-completing-read` but inside `org-ql-completing-read` the command is rebound to produce an `org-ql-view` buffer; and `org-ql-completing-read` also rebinds `embark-export` to do the same thing. This change has the following advantages:

1. Now `embark-export` just works and produces an appropriate org-ql-view buffer, but this does not introduce a dependency on embark since binding `(symbol-function 'embark-export)` is just a harmless no-op if embark is not loaded.

2. The functionality is still available for non-embark users and the key binding used to invoke it is configurable (by binding the pseudocommand `org-ql-export-view` to whatever the user prefers.

Let me know what you think. Also, I didn't put a lot of thought into the names for the new keymap and the new pseudocommand that gets rebound, so please change them if you think of better ones.

Reply:
Thanks, Omar, that's great!  I also remapped `embark-collect` to `embark-export` in that map, since `embark-collect` doesn't work with the results, so this way users don't have to worry about which one to use.